.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,6 +1,18 @@
 ## Proposed Commit Message
 <!-- Include a proposed commit message because all PRs are squash merged -->
 
+> summary: no more than 72 characters, ideally 50
+>
+> A description of what the change being made is and why it is being
+> made, if the summary line is insufficient.  The blank line above is
+> required. This should be wrapped at 72 characters, but otherwise has
```

Comment:
72 chars is fine.. but if you stop the subject at 72, then it will be longer than the body.

Reply:
Not 100% sure I follow this. (72 chars is the recommendation in [the git book](https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project).)

Reply:
If you say 'no more than 72 characters' for the subject, and github appends '(#XXXX)', then the subject line will have a max len of 79 chars in git.

79 is > 72, so your subject line will be longer than the body is allowed to be and by 10%,. that just seems odd.

Reply:
Aha, right, I see what you mean. I agree that it's a little odd.

My feeling is that the PR link being truncated in the GH UI isn't so much of a problem: it doesn't matter if you're just interested in the commit summary, and if you're wanting to click through to the PR then it adds 1 click to get there (at most; if even one character is in the displayed commit message then it gets linkified ([e.g.](https://github.com/canonical/cloud-init/commit/a6afe9c574681632c3c82d6bc8c6a7fb91425bdf)).

If we drop the summary limit to 70, then even with " (#12345)" appended, we still weigh in at 79 chars. How does that sound?